src/accordion/AccordionItem.tsx

```diff
@@ -1,84 +1,150 @@
-  TAccordionItemContext
->({
-  name: "useAccordionItemContext",
-  errorMessage:
-    "The `useAccordionItem` hook must be called from a descendent of the `AccordionItemProvider`.",
-  strict: true,
-});
+function useAccordionPanelId(options: AccordionItemOptions) {
+  return React.useMemo(
+    () =>
+      options.panels?.find(panel => panel.groupId === options.id)?.id ||
+      undefined,
+    [options.panels, options.id],
+  );
+}
 
 export const useAccordionItem = createHook<
   AccordionItemOptions,
   AccordionItemHTMLProps
 >({
-  name: "AccordionItem",
-  compose: unstable_useId,
+  name: "Accordion",
+  compose: [useButton, useCompositeItem],
   keys: ACCORDION_ITEM_KEYS,
 
+  useOptions({ focusable = true, ...options }) {
+    return { focusable, ...options };
+  },
+
   useProps(
-    { id, registerItem, activeItems, items, isOpen: isOpenOption },
-    { ref: htmlRef, children: htmlChildren, ...htmlProps },
+    options,
+    { onClick: htmlOnClick, onFocus: htmlOnFocus, ...htmlProps },
   ) {
-    const ref = React.useRef<HTMLElement>(null);
+    const selected = isAccordionSelected(options);
+    const accordionPanelId = useAccordionPanelId(options);
+    const onClickRef = useLiveRef(htmlOnClick);
+    const onFocusRef = useLiveRef(htmlOnFocus);
+
+    const onClick = React.useCallback(
+      (event: React.MouseEvent) => {
+        onClickRef.current?.(event);
+        if (event.defaultPrevented) return;
+        if (options.disabled) return;
+        if (!options.id) return;
+
+        if (selected) {
+          if (options.allowToggle && !options.allowMultiple) {
+            // Do not send null to make the toggle because that will also reset
+            //  the current Id in composite hence thats handled directly in state
+            options.select(options.id);
+            return;
+          }
+
+          options.unSelect(options.id);
+          return;
+        }
 
-    React.useLayoutEffect(() => {
-      if (!id) return undefined;
+        options.select?.(options.id);
+      },
 
-      registerItem?.({ id, ref });
-    }, [id, registerItem]);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [options.disabled, selected, options.select, options.id],
+    );
+
+    const onFocus = React.useCallback(
+      (event: React.FocusEvent) => {
+        onFocusRef.current?.(event);
+        if (event.defaultPrevented) return;
+        if (options.disabled) return;
+        if (options.manual) return;
+        if (!options.id) return;
+
+        if (selected) {
+          if (options.allowToggle && !options.allowMultiple) {
+            // Do not send null to make the toggle because that will also reset
+            //  the current Id in composite hence thats handled directly in state
+            options.select(options.id);
+            return;
+          }
+
+          options.unSelect(options.id);
+          return;
+        }
+
+        options.select?.(options.id);
```

Comment:
This bit of code is same as the OnClick handler's, extract it out.

Reply:
I thought you will say this 😆

Reply:
Yeah lol 😆 My eyes & brain screams when they see repetitive code which can be refactored :p